lib/chef/resource/aws_internet_gateway.rb

```diff
@@ -0,0 +1,25 @@
+require 'chef/resource/aws_resource'
+require 'chef/provisioning/aws_driver'
+
+class Chef::Resource::AwsInternetGateway < Chef::Resource::AwsResource
+  self.resource_name = 'aws_internet_gateway'
+  self.databag_name = 'aws_internet_gateway'
+
+  actions :create, :delete, :attach, :nothing
+  default_action :create
+
+  attribute :name, kind_of: String, name_attribute: true
```

Comment:
I believe you can omit this line - all resources by default have a `:name` attribute.  And I don't think we want to encourage resources that look like
```ruby
aws_internet_gateway "my_name" do
  name "my_name"
end
```